src/Microsoft.DotNet.MSBuildSdkResolver/EnvironmentProvider.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.DotNet.MSBuildSdkResolver
+{
+    internal class EnvironmentProvider
+    {
+        private IEnumerable<string> _searchPaths;
+        private IEnumerable<string> _executableExtensions;
+
+        private readonly Func<string, string> _getEnvironmentVariable;
+
+        public EnvironmentProvider(Func<string, string> getEnvironmentVariable)
+        {
+            _getEnvironmentVariable = getEnvironmentVariable;
+        }
+
+        public IEnumerable<string> ExecutableExtensions
+        {
+            get
+            {
+                if (_executableExtensions == null)
+                {
+
+                    _executableExtensions = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                        ? _getEnvironmentVariable("PATHEXT")
```

Comment:
I don't think we should consider dotnet.bat on path as a candidate. We should search for dotnet.exe on Windows and dotnet elsewhere.